packages/compass-sidebar/src/modules/data-service.ts

```
@@ -1,46 +1,21 @@
-import type { DataService } from '@mongodb-js/compass-connections/provider';
-import type { RootAction, SidebarThunkAction } from '.';
-
-export const SET_DATASERVICE = 'sidebar/SET_DATASERVICE' as const;
-interface SetDataServiceAction {
-  type: typeof SET_DATASERVICE;
-  dataService: DataService;
-}
-export const SET_CSFLE_ENABLED = 'sidebar/SET_CSFLE_ENABLED' as const;
-interface SetCSFLEEnabledAction {
-  type: typeof SET_CSFLE_ENABLED;
-  enable: boolean;
-}
-export type DataServiceAction = SetDataServiceAction | SetCSFLEEnabledAction;
-export type DataServiceState = DataService | null;
-
-export const INITIAL_STATE: DataServiceState = null;
-
-export default function reducer(
-  state: DataServiceState = INITIAL_STATE,
-  action: RootAction
-): DataServiceState {
-  if (action.type === SET_DATASERVICE) {
-    return action.dataService;
-  }
-  if (action.type === SET_CSFLE_ENABLED) {
-    state?.setCSFLEEnabled(action.enable);
-  }
-  return state;
-}
-
-export const setDataService = (
-  dataService: DataService
-): SetDataServiceAction => ({
-  type: SET_DATASERVICE,
-  dataService,
-});
+import type { SidebarThunkAction } from '.';
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
 
 export const setConnectionIsCSFLEEnabled = (
+  connectionId: ConnectionInfo['id'],
   enable: boolean
-): SidebarThunkAction<void, SetCSFLEEnabledAction> => {
-  return (dispatch, _getState, { globalAppRegistry }) => {
-    dispatch({ type: SET_CSFLE_ENABLED, enable });
+): SidebarThunkAction<void, never> => {
+  return (_dispatch, _getState, { globalAppRegistry, connectionsManager }) => {
+    const dataService =
+      connectionsManager.getDataServiceForConnection(connectionId);
+
+    if (!dataService) {
+      // This should be unreachable because this is only visible when
+      // we are connected.
+      return;
```

wouldn't it be safer to throw then? to avoid a silent failure in the future?